### PR TITLE
librbd: Support for shared locking in ManagedLock

### PR DESCRIPTION
**Pull request by rjfd**
Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

**Comment by dillaman** [comment]
Note: the GetLockerRequest state machine will need to be updated after #12922 is merged

**Comment by dillaman** [comment]
@rjfd nit: in your commit messages, you should use a "librbd"-prefix when making changes to librbd. The "rbd"-prefix is mostly used for the rbd CLI tool.

File: src/librbd/ManagedLock.h [hunk 1]
```diff
@@ -29,13 +29,19 @@ class ManagedLock {
 public:
   static const std::string WATCHER_LOCK_TAG;
 
+  enum Mode {
```
Comment (dillaman) [inline src/librbd/ManagedLock.h]: Nit: can you move this to managed_lock/Types.h? I have assigned myself a tracker ticket to remove all unnecessary dependencies to this templated class in order to clean up the unit tests.

**Comment by rjfd** [comment]
@dillaman addressed the comments and pushed new changes

**Comment by dillaman** [comment]
@rjfd Rebase is now required

**Comment by rjfd** [comment]
@dillaman rebase done

**Pull request merged**